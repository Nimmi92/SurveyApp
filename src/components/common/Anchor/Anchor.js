import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const StyledA = styled.a`
 font-size: 0.8em;
 margin-bottom: 10px;
 color: #24292e;
 display: inline-block;

 &:hover {
  color: #0366d6;
 }
`
const Anchor = (props) => {

  return (
    <StyledA href={props.href} onClick={props.onClick}>
    	{props.linkText}
    </StyledA>
  );
};


const { string, func } = PropTypes;

Anchor.propTypes = {
	linkText: string,
	onClick: func,
	href: string,
}

export default Anchor;