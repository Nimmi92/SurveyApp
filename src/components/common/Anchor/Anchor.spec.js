import React from 'react';
import { shallow } from 'enzyme';
import Anchor from './Anchor';
import toJson from "enzyme-to-json";

describe('<Anchor />', () => {
  it('should be an anchor tag', () => {
    const props = {
      href: 'www.google.com',
      linkText: 'Google',
      onClick: jest.fn()
    };

    const wrapper = shallow(<Anchor {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

})