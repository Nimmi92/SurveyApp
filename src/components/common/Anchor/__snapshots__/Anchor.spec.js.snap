// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Anchor /> should be an anchor tag 1`] = `
<ForwardRef
  href="www.google.com"
  onClick={[MockFunction]}
>
  Google
</ForwardRef>
`;
